### PR TITLE
[Major Release: 3.6.0]: Automated PR for new major release updates

### PR DESCRIPTION
**Pull request by ide-developer**
Automated release kickoff run for release **3.6.0** using:

- optional `versions_config.yml` overrides from workflow inputs
- `make kickoff-release` (build-args sync, lock refresh, ODH/RHDS RPM locks, rollout)
- `make test` validation gate inside kickoff-release

### CODE REVIEW

**Comment by atheo89** [comment]
@coderabbitai full review

**Comment by jiridanek** [comment]
@coderabbit review --dir jupyter manifests prefetch-input runtimes

**Review by jiridanek — APPROVED** [review]
tl;dr

**Comment by jiridanek** [comment]
Retriggering failed Konflux builds:

/build-runtime-pytorch-llmcompressor
/build-datascience
/build-minimal-cuda
/build-pytorch-cuda

**Review by ide-developer — COMMENTED** [review]
## Review plan (executed)

1. **Scope & intent** — confirm this is the automated 3.6.0 kickoff (`make kickoff-release`) and enumerate changed artifact classes.
2. **Release mechanics** — verify `versions_config.yml` → `Makefile` → `build-args` → lockfiles → manifests propagation.
3. **Manifest rollout** — check N / N-1 tag shift, commit placeholders, and ODH vs RHOAI parity.
4. **CI signal** — triage merge gates, GHA matrix, Konflux, and `validation-of-sw-versions-in-imagestreams`.
5. **Verdict** — classify blockers vs flakes and state merge readiness.

---

## Summary

Automated release kickoff for **3.6.0** looks mechanically correct: `full_version` 3.5.0 → 3.6.0, `RELEASE` default bumped, build-args / pylock / RPM locks refreshed, and imagestream manifests rolled forward (new `3.6` N-tag, prior `3.5` demoted to N-1). Merge gates (`check-generated-code`, `pytest-tests`, `go-tests`, `prek`, `validation-of-params-env-*`) are green.

**Not merge-ready yet** — three classes of red checks remain, one of which is a real manifest-validation blocker.

## Findings

### Blocker — `validation-of-sw-versions-in-imagestreams`

`test_old_tag_annotations_match_quay` fails for **tag 3.5** on both ODH and RHOAI manifests:

| Imagestream | Failure |
|-------------|---------|
| `code-server-notebook` | `jupyterlab` and `codeflare-sdk` present in published image but missing from manifest annotations |
| `jupyter-rocm-{minimal,pytorch,tensorflow}` | Manifest lists **ROCm 7.14** but Clair/Quay scan of the published 3.5 digest does not contain it |
| RHOAI ROCm trio (additional) | Clair response empty / fetch failures on published digests |

The kickoff correctly shifted N→N-1, but the **3.5 slot annotations now describe 3.6-era package versions** while the images referenced by those digests are still the previously published 3.5 builds (ROCm 6.4 era, incomplete codeserver annotation set). This must be reconciled before merge — either back-annotate 3.5 slots to match published images, or wait until GA 3.5 images are rebuilt and re-tagged with matching SHAs.

### In progress — Konflux (4 retriggered)

Initial Konflux failures on datascience, minimal-cuda, pytorch-cuda, and runtime-pytorch-llmcompressor were retriggered individually; replacements are queued/in progress. Several other Konflux pipelines still running from the original PR event.

### Likely flakes — GHA matrix

| Job | Note |
|-----|------|
| `runtime-minimal · linux/amd64 [odh]` | Failed on PR workflow; **passed** on push workflow and RHOAI lane |
| `jupyter-minimal · linux/ppc64le / odh` | Failed on push workflow; **passed** on PR workflow |

Worth a targeted re-run if they stay red after Konflux settles; not indicative of a release-bump defect.

### Follow-up (non-blocking for ODH, tracked)

`manifests/rhoai/base/commit.env` carries a `TODO` to refresh `*-commit-3-5` hashes after RHOAI 3.5 GA — expected for cross-stream kickoff, but should be closed before RHOAI-side consumption.

## Verdict

**Comment / hold** — release mechanics are sound, but merge should wait for:
1. Green `validation-of-sw-versions-in-imagestreams` (manifest ↔ Quay alignment for 3.5 slots)
2. All Konflux PR pipelines green
3. Confirmation GHA flakes are retried or explained

No functional code changes beyond housekeeping; risk is concentrated in manifest annotation accuracy for historical tags.

File: manifests/odh/base/code-server-notebook-imagestream.yaml [hunk 1]
```diff
@@ -44,37 +44,39 @@ spec:
...
     - annotations:
         # language=json
         opendatahub.io/notebook-software: |
           [
-            {"name": "code-server", "version": "4.104"},
+            {"name": "code-server", "version": "4.106"},
             {"name": "Python", "version": "v3.12"}
           ]
         # language=json
         opendatahub.io/notebook-python-dependencies: |
```
Comment (ide-developer) [inline manifests/odh/base/code-server-notebook-imagestream.yaml:59]: **Finding (medium):** `validation-of-sw-versions-in-imagestreams` reports `jupyterlab` and `codeflare-sdk` installed in the published **3.5** codeserver image but absent from these `notebook-python-dependencies` annotations. The kickoff rolled forward dependency versions (MLflow/Feast added) without carrying forward packages that Clair still detects in the live 3.5 digest.

Action: align the 3.5 (N-1) annotation block with the actual published image contents, or update the image reference to a digest that matches the new annotation set.

File: manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml [hunk 2]
```diff
@@ -34,28 +34,28 @@ spec:
...
+      name: "3.6"
       referencePolicy:
         type: Source
     # N - 1 Version of the image
     - annotations:
         # language=json
         opendatahub.io/notebook-software: |
           [
-            {"name": "ROCm", "version": "6.4"},
+            {"name": "ROCm", "version": "7.14"},
```
Comment (ide-developer) [inline manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml:45]: **Finding (medium):** Manifest lists `ROCm 7.14` for tag **3.5**, but `test_old_tag_annotations_match_quay` could not find ROCm 7.14 in the published Quay/Clair scan for that digest. Same pattern on pytorch-rocm and tensorflow-rocm imagestreams.

This suggests N-1 annotations inherited 3.6-target versions while still pointing at pre-kickoff 3.5 image SHAs. Verify whether 3.5 slots should retain 6.4-era annotations until GA images are republished.

File: manifests/rhoai/base/commit.env [hunk 3]
```diff
@@ -1,3 +1,17 @@
+# TODO: Update the hashes after 3.5 official release, these values fetched from
```
Comment (ide-developer) [inline manifests/rhoai/base/commit.env:1]: **Finding (low):** Placeholder `*-commit-3-5=955afa6` entries with an explicit TODO to refresh after RHOAI 3.5 GA. Acceptable for ODH kickoff timing, but track closure before downstream RHOAI manifest sync consumes these values.

File: codeserver/ubi9-python-3.12/build-args/cpu.conf
Comment (jiridanek) [inline codeserver/ubi9-python-3.12/build-args/cpu.conf:1]: @coderabbitai recheck, verify, elaborate critically

<details>

Review plan (executed)
Scope & intent — confirm this is the automated 3.6.0 kickoff (make kickoff-release) and enumerate changed artifact classes.
Release mechanics — verify versions_config.yml → Makefile → build-args → lockfiles → manifests propagation.
Manifest rollout — check N / N-1 tag shift, commit placeholders, and ODH vs RHOAI parity.
CI signal — triage merge gates, GHA matrix, Konflux, and validation-of-sw-versions-in-imagestreams.
Verdict — classify blockers vs flakes and state merge readiness.
Summary
Automated release kickoff for 3.6.0 looks mechanically correct: full_version 3.5.0 → 3.6.0, RELEASE default bumped, build-args / pylock / RPM locks refreshed, and imagestream manifests rolled forward (new 3.6 N-tag, prior 3.5 demoted to N-1). Merge gates (check-generated-code, pytest-tests, go-tests, prek, validation-of-params-env-*) are green.

Not merge-ready yet — three classes of red checks remain, one of which is a real manifest-validation blocker.

Findings
Blocker — validation-of-sw-versions-in-imagestreams
test_old_tag_annotations_match_quay fails for tag 3.5 on both ODH and RHOAI manifests:

Imagestream	Failure
code-server-notebook	jupyterlab and codeflare-sdk present in published image but missing from manifest annotations
jupyter-rocm-{minimal,pytorch,tensorflow}	Manifest lists ROCm 7.14 but Clair/Quay scan of the published 3.5 digest does not contain it
RHOAI ROCm trio (additional)	Clair response empty / fetch failures on published digests
The kickoff correctly shifted N→N-1, but the 3.5 slot annotations now describe 3.6-era package versions while the images referenced by those digests are still the previously published 3.5 builds (ROCm 6.4 era, incomplete codeserver annotation set). This must be reconciled before merge — either back-annotate 3.5 slots to match published images, or wait until GA 3.5 images are rebuilt and re-tagged with matching SHAs.

In progress — Konflux (4 retriggered)
Initial Konflux failures on datascience, minimal-cuda, pytorch-cuda, and runtime-pytorch-llmcompressor were retriggered individually; replacements are queued/in progress. Several other Konflux pipelines still running from the original PR event.

Likely flakes — GHA matrix
Job	Note
runtime-minimal · linux/amd64 [odh]	Failed on PR workflow; passed on push workflow and RHOAI lane
jupyter-minimal · linux/ppc64le / odh	Failed on push workflow; passed on PR workflow
Worth a targeted re-run if they stay red after Konflux settles; not indicative of a release-bump defect.

Follow-up (non-blocking for ODH, tracked)
manifests/rhoai/base/commit.env carries a TODO to refresh *-commit-3-5 hashes after RHOAI 3.5 GA — expected for cross-stream kickoff, but should be closed before RHOAI-side consumption.

Verdict
Comment / hold — release mechanics are sound, but merge should wait for:

Green validation-of-sw-versions-in-imagestreams (manifest ↔ Quay alignment for 3.5 slots)
All Konflux PR pipelines green
Confirmation GHA flakes are retried or explained
No functional code changes beyond housekeeping; risk is concentrated in manifest annotation accuracy for historical tags.

manifests/odh/base/code-server-notebook-imagestream.yaml
            {"name": "Python", "version": "v3.12"}
          ]
        # language=json
        opendatahub.io/notebook-python-dependencies: |
@ide-developer
ide-developer
1 minute ago
Member
Author
Finding (medium): validation-of-sw-versions-in-imagestreams reports jupyterlab and codeflare-sdk installed in the published 3.5 codeserver image but absent from these notebook-python-dependencies annotations. The kickoff rolled forward dependency versions (MLflow/Feast added) without carrying forward packages that Clair still detects in the live 3.5 digest.

Action: align the 3.5 (N-1) annotation block with the actual published image contents, or update the image reference to a digest that matches the new annotation set.

@jiridanek	Reply...
manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml
        opendatahub.io/notebook-software: |
          [
            {"name": "ROCm", "version": "6.4"},
            {"name": "ROCm", "version": "7.14"},
@ide-developer
ide-developer
1 minute ago
Member
Author
Finding (medium): Manifest lists ROCm 7.14 for tag 3.5, but test_old_tag_annotations_match_quay could not find ROCm 7.14 in the published Quay/Clair scan for that digest. Same pattern on pytorch-rocm and tensorflow-rocm imagestreams.

This suggests N-1 annotations inherited 3.6-target versions while still pointing at pre-kickoff 3.5 image SHAs. Verify whether 3.5 slots should retain 6.4-era annotations until GA images are republished.

@jiridanek	Reply...
manifests/rhoai/base/commit.env
@@ -1,3 +1,17 @@
# TODO: Update the hashes after 3.5 official release, these values fetched from
@ide-developer
ide-developer
1 minute ago
Member
Author
Finding (low): Placeholder *-commit-3-5=955afa6 entries with an explicit TODO to refresh after RHOAI 3.5 GA. Acceptable for ODH kickoff timing, but track closure before downstream RHOAI manifest sync consumes these values.

@jiridanek	Reply...

</details>

File: manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml (hunk 2 above)
Comment (jiridanek) [reply manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml:45]: @atheo89 , @coderabbitai looks legit, maybe better to fix in subsequent PR, though

File: manifests/odh/base/code-server-notebook-imagestream.yaml (hunk 1 above)
Comment (jiridanek) [reply manifests/odh/base/code-server-notebook-imagestream.yaml:59]: @coderabbitai I don't understand, explore more and explain clearly

File: manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml (hunk 2 above)
Comment (atheo89) [reply manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml:45]: agree

Comment (atheo89) [reply manifests/odh/base/jupyter-rocm-minimal-notebook-imagestream.yaml:45]: Hey, I checked it today, the 3.5 ROCm images are based on 7.14 
```
podman run --rm --platform=linux/amd64 --entrypoint="" \
  quay.io/opendatahub/odh-workbench-jupyter-minimal-rocm-py312-ubi9@sha256:a09fae755bfb8f854b9cefc6bd333572d5d59b93a0a322a7c7fb9b1b684b4f1f \
  printenv ROCM_VERSION
```

The test was failing because was looking in wrong detection path. 
This PR https://github.com/opendatahub-io/notebooks/pull/4284 updates it - now the test pass for the ROCm findings (but it still fails with old issues (i will not fix them now)
 https://github.com/opendatahub-io/notebooks/actions/runs/30893982206/job/91942383694?pr=4284
